src/gui/main_window.cpp

```diff
@@ -1132,7 +1143,7 @@ bool MainWindow::showSaveAsDialog()
 	// Fails when using different formats for import and export:
 	//	Q_ASSERT(FileFormats.findFormatForFilename(path) == format);
 	
-	return savePath(path);
+	return savePath(path, format);
```

Comment:
Hint: This is the root of the changes: We have got the format from the file dialog's *filter* and must pass it down.